Samples/Excel.OfflineStorageAddin/README.md

```diff
@@ -0,0 +1,126 @@
+
+## Key parts of this sample
+
+Navigate to *Excel.OfflineStorageAddin/src/taskpane/taskpane.js* to find the implementation of `localStorage` described below. 
+
+### Implementing `localStorage` to offline data
+The *Excel.OfflineStorageAddin/src/taskpane/taskpane.js* file contains the `loadTable()` function, that uses `localStorage` to display a table of fictitious basketball player stats when a user loses connection.
+
+In the sample code, the `loadTable()` function first attempts to perform an AJAX call to retrieve information from  the offline data file, `sampleData.json`. In this example, the data from our local file does not ever change; however, performing the AJAX call allows the add-in to always retrieve the most up to date information. If the AJAX call is successful, the function passes the data returned from the file to `createTable()` to produce a table. Before being stored into `localStorage`, the data is also converted into a JSON object so that it can be easily parsed when `localStorage` is accessed. This process is shown in the following code:
+
+```js
+$.ajax({
+    dataType: "json",
+    url: "sampleData.json",
+    success: function (result, status, xhr) {
+        // Stores the JSON retrieved from the AJAX call as a string in
+        // local storage under the key "PlayerData"
+        localStorage.PlayerData = JSON.stringify(result);
+
+        // Sends the new data to the table
+        createTable(result);
+    },
+```
+If the function fails to successfully get the data from the AJAX call, the function first checks if the basketball player data was previously cached into local storage, as shown in the code below. If it exists, the data is parsed from JSON into a readable text format before being passed to `createTable()`, a function which creates a table from the given data. Otherwise, the function returns an error back to the console.
+
+```js
+error: function (xhr, status, error) {
+    // If the connections fails, checks if "PlayerData" was previously stored in local storage
+    if (localStorage.PlayerData) {
+        // Retrieves the string saved earlied under the key "PlayerData"
+        // and parses it into an object
+        let dataObject = JSON.parse(localStorage.PlayerData);
+
+        // Sends the saved data to the table
+        createTable(dataObject);
+    }
+    else {
+        console.log("Player data failed to load with error: " + error);
+    }
+}
+```
+
+## Security notes
+
+In the webpack.config.js file, a header is set to  `"Access-Control-Allow-Origin": "*"`. This is only for development purposes. You should lock this header down to only allowed domains in production code.
+

```

Comment:
The certificates are not **self-signed**. `office-addin-dev-certs` creates a SSL certificate that is signed by CA certificate and will prompt to install the CA certificate so that the SSL certificate is trusted.

Reply:
Fixed